src/Storages/StorageDistributed.cpp

```diff
@@ -314,6 +314,10 @@ std::optional<QueryProcessingStage::Enum> getOptimizedQueryProcessingStage(const
     if (select.group_by_with_totals || select.group_by_with_rollup || select.group_by_with_cube)
         return {};
 
+    // Window functions are not supported.
```

Comment:
Is this both for `distributed_group_by_no_merge` = 1 and 2, or only for 2? For 1 window functions should work. Not really sure what this function does...

Reply:
This is realization of `optimize_distributed_group_by_sharding_key`, and it simply prohibit this optimization if there is window functions in the query.